dbms/src/Common/LooseBoundedMPMCQueue.h

```diff
@@ -230,11 +235,16 @@ class LooseBoundedMPMCQueue
     }
 
     template <typename U>
-    ALWAYS_INLINE void pushFront(U && data)
+    ALWAYS_INLINE MPMCQueueResult pushFront(U && data)
     {
         Int64 memory_usage = get_auxiliary_memory_usage(data);
         queue.emplace_front(std::forward<U>(data), memory_usage);
         current_auxiliary_memory_usage += memory_usage;
+        if (push_callback)
+        {
+            if (!push_callback(queue.front().data))
+                return MPMCQueueResult::CANCELLED;
```

Comment:
Is there a need to call `cancel()` here?

Reply:
Done